processor/resourcedetectionprocessor/internal/aws/ec2/documentation.md

```diff
@@ -8,6 +8,7 @@
 
 | Name | Description | Values | Enabled |
 | ---- | ----------- | ------ | ------- |
+| aws.ec2.instance_life_cycle | The EC2 instance life cycle | Any Str | true |
```

Comment:
Is this a state like attribute? I wonder if that should be a metric then 🤔 .

See https://github.com/open-telemetry/semantic-conventions/issues/1554.

Probably discussing this and defining this first with Semantic Conventions project would help.

Reply:
@ChrsMark does it make sense to you, can we move forward this way?

Reply:
Sure, the issue has been created. https://github.com/open-telemetry/semantic-conventions/issues/2300

BTW, is e2e required for this change? I've check the test code, since most of them are generated, I'm not sure if I sure do some e2e test, if so where can I find the guideline how to do it?

Reply:
> Sure, the issue has been created. https://github.com/open-telemetry/semantic-conventions/issues/2300. Apart from aws.ec2.instance_life_cycle, I've also added aws.ec2.vpc_id in the proposal, implementation PR will be created later.

Thank's! @dashpole @Aneurysm9 please consider this too as part of this enhancement.

Reply:
Yeah, I agree. 'instance_life_cycle' is not generic enough, that's why it's under 'aws.ec2' namespace.